code/modules/projectiles/wavegun.dm

```diff
@@ -30,34 +32,60 @@ toxic - poisons
 */
 	damage_type = D_ENERGY
 	//With what % do we hit mobs laying down
-	hit_ground_chance = 20
+	hit_ground_chance = 33
 	//Can we pass windows
 	window_pass = 0
 
-	on_pointblank(var/obj/projectile/P, var/mob/living/M)
-		stun_bullet_hit(P, M)
 
-/datum/projectile/wavegun/burst
-	shot_number = 3
-	power = 50
-	cost = 75
-	ks_ratio = 0.0
-	damage_type = D_ENERGY
-	sname = "burst wave"
-
-/datum/projectile/wavegun/paralyze
-	icon_state = "wavegun-2"
+/datum/projectile/wavegun/transverse //expensive taser shots that go through /everything/
 	shot_number = 1
-	power = 40
-	ks_ratio = 0.0
+	power = 10 //half the power of a taser at range 1-3, delivers a nasty punch at the 4-tile sweetspot
+	dissipation_delay = 4
+	dissipation_rate = -30
+	max_range = 5 //super short. about 4 tile max range
 	cost = 50
+	hit_ground_chance = 100 //no escape
+	pierces = -1 //no limits
+	goes_through_walls = 1
+	window_pass = 1
+	ticks_between_mob_hits = 1
 	damage_type = D_ENERGY
-	sname = "paralysis wave"
+	sname = "transverse wave"
+	icon_state = "wave-g"
+		
 
-	on_hit(atom/hit)
-		if(isliving(hit))
-			var/mob/living/L = hit
-			L.changeStatus("stunned", 10 SECONDS)
 
-	on_pointblank(var/obj/projectile/P, var/mob/living/M)
-		M.changeStatus("stunned", 10 SECONDS)
+		
+
+/datum/projectile/wavegun/emp
+	shot_number = 1
+	power = 0
+	dissipation_delay = 1
+	dissipation_rate = -10 //only reliable past a few tiles 
+	max_range = 18 //taser-and-a-half range
+	cost = 100 //two shots, unless you upgrade to a pulserifle/etc cell
+	hit_ground_chance = 0
+	damage_type = D_SPECIAL
+	sname = "electromagnetic distruption wave"
+	icon_state = "wave-emp"
+	disruption = 25
+
+	on_hit(atom/H, angle, var/obj/projectile/P)
+		var/turf/T = get_turf(H)
+		if(prob(P.power*2.5))
+			for(var/atom/O in T.contents)
+				O.emp_act()
+		if(P.power >= 100) //10 tile shot, wow
+			for(var/turf/tile in range(1,T))
+				for(var/atom/O in tile.contents)
+					O.emp_act()
```

Comment:
You probably don't intend on applying `emp_act()` to `T`'s contents twice, right? Also, you probably wanna check for mobs and items before calling `emp_act()`, since they're the types that actually have functionality for this proc.

Reply:
The double-emp was actually intentional, as a relic of earlier revision on the concept where it called multiple emp_act()s depending on range (for the purposes of shooting borgs/doors/etc). I'd prefer to leave it in and change it if it becomes an issue in gameplay.
Going to also note that other EMPs don't have checking, so probably those should be cleaned up in a future patch